cpp/src/arrow/vendored/fast_float/README.md

```diff
@@ -20,7 +20,7 @@
 # fast_float
 
 The files in this directory are vendored from fast_float
-git tag `v3.8.1`.
+git tag `v3.10.1`.
```

Comment:
Do you really want to use 3.10.1?
It seems that the latest version is 5.2.0: https://github.com/fastfloat/fast_float/releases

Reply:
after 3.10.1 they started using c++ 20 and it cause a lot of build errors. In reality - we just need in parse_number.h 
```
#ifdef FASTFLOAT_ALLOWS_LEADING_PLUS // disabled by default
  if (*first == UC('+')) {
      ++first;
  }
```

Reply:
OK. Could you describe it in descriptions of the PR and the associated issue?